src/ExchangeSharp/API/Exchanges/Kraken/ExchangeKrakenAPI.cs

```diff
@@ -749,13 +749,17 @@ protected override async Task<IEnumerable<MarketCandle>> OnGetCandlesAsync(strin
 
 		protected override async Task<ExchangeOrderResult> OnPlaceOrderAsync(ExchangeOrderRequest order)
 		{
+			IEnumerable<ExchangeMarket> markets = await OnGetMarketSymbolsMetadataAsync();
+			ExchangeMarket market = markets.Where(m => m.MarketSymbol == order.MarketSymbol).First<ExchangeMarket>();
+
 			object nonce = await GenerateNonceAsync();
 			Dictionary<string, object> payload = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase)
 			{ { "pair", order.MarketSymbol }, { "type", (order.IsBuy ? "buy" : "sell") }, { "ordertype", order.OrderType.ToString().ToLowerInvariant() }, { "volume", order.RoundAmount().ToStringInvariant() }, { "trading_agreement", "agree" }, { "nonce", nonce }
 			};
 			if (order.OrderType != OrderType.Market)
 			{
-				payload.Add("price", order.Price.ToStringInvariant());
+				int precision = BitConverter.GetBytes(Decimal.GetBits((decimal)market.PriceStepSize)[3])[2];
+				payload.Add("price", Math.Round(order.Price, precision).ToStringInvariant());
```

Comment:
This rounding avoids an error returned by Kraken's API that `price` can only have so many decimals.